src/layouts/sidebar-sidecar/utils/product-url-adjusters.ts

```diff
@@ -52,18 +52,50 @@ function prefixSentinelDocsUrls(inputUrl: string): string {
   if (isAbsoluteUrl(inputUrl)) return inputUrl
   // We now know we're dealing with internal URLs only
   let outputUrl = inputUrl.slice()
-  // First, we have to remove the `/sentinel` prefix from all URLs that have it,
-  // as we re-add the product prefix in "components/docs-anchor"
+  // We gather some information on what link routes we're dealing with.
+  // Note that on the client-side, "components/docs-anchor" will modify
+  // URLs that start with any of the "basePath" strings defined in
+  // the "basePaths" array in "data/{product.json".
+  // Any such URLs will be prefixed with the product name, in this case
+  // "sentinel". However, unlike other product sites, links in the Sentinel
+  // docs already contain this prefix. So, we need to remove the prefix
+  // from any URLs that will later be modified based on "basePaths", while
+  // retaining it for URLs that will not match "basePaths" (namely
+  // "downloads" and "changelog").
+  // TODO: could this be simplified by doing all URL manipulations here
+  // TODO: at server time, rather than also prefixing some URLs on the
+  // TODO: client-side using "components/docs-anchor"?
+  // TODO: (May be useful longer term as well, eg if authors become
+  // TODO: accustomed to the new URL structure on dev-dot, and want to write
+  // TODO: with those URLs in mind, rather than the old dot-io URL structures,
+  // TODO: then we could apply these remark transformations to all content
+  // TODO: and remove the need for prefixing and manipulating.)
+  const isBasePathExceptDocs = sentinelData.basePaths.reduce(
+    (wasMatched, basePath) => {
+      const isMatch =
+        outputUrl == `/sentinel/${basePath}` ||
+        outputUrl.startsWith(`/sentinel/${basePath}/`)
+      return wasMatched || isMatch
+    },
+    false
+  )
+  // Note that "docs" links will not be caught by the above check,
+  // as on the sentinel site they are found under `/sentinel/*`, rather
+  // than under `/sentinel/docs` as defined by our basePaths here in dev-dot.
+  // So, we assume that if an internal link prefixed with `/sentinel`
+  // it NOT one of the other two known links (downloads and changelog),
+  // then it must be a docs link.
+  const isDownloadsUrl = outputUrl == '/sentinel/downloads'
+  const isChangelogUrl = outputUrl == '/sentinel/changelog'
```

Comment:
😅

Reply:
Yeah I really don't love this - very open to other suggestions here 🙇‍♂️

I think one step to clarity here might be moving the `adjustedHref` work we're doing in `components/docs-anchor` into here - then for sentinel at least, we're not taking out the `/sentinel` prefix, but only for routes for which we've specified `basePaths` in `data/sentinel.json`, only to re-add it on the client.

This is maybe Sentinel-specific, but it got me thinking about how URLs are changing as we move to dev-portal, and that once we are fully migrated, authors will begin thinking of these URLs in the dev-portal context but will still be authoring them in the (now inaccurate) "dot-io" context.  The above step of consolidating all URL manipulation into a `remark` plugin could also open the door to running a transform on all content in order to update the URLs to actually reflect the URLs in dev-portal. Once that's done, and the content at the source is correct, we could dispense with these `product-url-adjusters` entirely. But... probably won't make sense to anyone but us until after we've launched dev-dot for all products 🤷‍♂️

Reply:
> we could dispense with these product-url-adjusters entirely

I'm not sure if this is true because of versioned docs. Unless we go update every version of docs? 🤔